homeassistant/components/melcloud/__init__.py

```diff
@@ -66,7 +66,15 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Establish connection with MELClooud."""
     conf = entry.data
-    mel_devices = await mel_devices_setup(hass, conf[CONF_TOKEN])
+    try:
+        mel_devices = await mel_devices_setup(hass, conf[CONF_TOKEN])
```

Comment:
This can on it's own raise `ConfigEntryNotReady`.
either fix `mel_devices_setup` or include `except ConfigEntryNotReady` here

Reply:
I don't think `mel_devices_setup()` is used elsewhere so in that case I would think all exceptions should be caught here and you can clean it out from `mel_devices_setup()`

Reply:
Should be fixed now, right?